### DISCUSSION TITLE
Elsa.WriteLine expression and memory reference

### DISCUSSION THREAD

**Discussion by mremadevi** [elsa-workflows/elsa-core#5300] [Q&A]
How to set memory reference for Elsa.Writeline activity with input set through expressions?  Elsa studio 3.2 has option to set input through expressions, but the generated JSON doesn`t have memory reference assigned

{
  "text": {
    "typeName": "String",
    "expression": {
      "type": "JavaScript",
      "value": "`Nice to meet you, ${getName()}!`"
    },
    "memoryReference": {
      "id": "WriteLine2:input-1"
    }
  },
  "id": "WriteLine2",
  "type": "Elsa.WriteLine"
}

**Comment by glime-ai** [comment]
To set a memory reference for an `Elsa.WriteLine` activity with input set through expressions in Elsa Studio 3.2, you would typically not need to manually set the memory reference in the JSON. The memory reference is an internal mechanism used by Elsa to track the state of inputs and outputs across activities within a workflow.

When you use the visual designer to create or edit a workflow, the designer interface should handle the creation of these memory references automatically. If you are programmatically creating a workflow and need to set the memory reference, you would do so by assigning a unique identifier to the `memoryReference` property within the activity's JSON representation.

However, if you are encountering an issue where the memory reference is not being generated correctly, this could be a bug or a missing feature in the version of Elsa Studio you are using. In such a case, you should check the official documentation for any updates or report the issue on the Elsa GitHub repository.

If you are manually editing the JSON and need to add a memory reference, you can follow the structure shown in your example, ensuring that the `id` is unique within the workflow:

```json
{
  "text": {
    "typeName": "String",
    "expression": {
      "type": "JavaScript",
      "value": "`Nice to meet you, ${getName()}!`"
    },
    "memoryReference": {
      "id": "UniqueIdentifierForMemoryReference"
    }
  },
  "id": "WriteLine2",
  "type": "Elsa.WriteLine"
}
```

Replace `UniqueIdentifierForMemoryReference` with an actual unique identifier for your workflow.

If you continue to have issues, please provide more details about the steps you are taking and the specific problem you are encountering, so I can assist you further.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
